runtime/common/src/lib.rs

```diff
@@ -242,6 +228,67 @@ pub mod parachains {
 	}
 }
 
+/// CurrencyIdConvert
+/// This type implements conversions from our `CurrencyId` type into `MultiLocation` and vice-versa.
+/// A currency locally is identified with a `CurrencyId` variant but in the network it is identified
+/// in the form of a `MultiLocation`, in this case a pCfg (Para-Id, Currency-Id).
+pub struct CurrencyIdConvert<ParachainId, AssetRegistry>(sp_std::marker::PhantomData<(ParachainId, AssetRegistry)>);
+
+impl<ParachainId: Get<ParaId>, AssetRegistryInspector: AssetRegistryInspect> Convert<CurrencyId, Option<MultiLocation>> for CurrencyIdConvert<ParachainId, AssetRegistryInspector> {
+	fn convert(id: CurrencyId) -> Option<MultiLocation> {
+		AssetRegistryInspector::metadata(&id)
+			.filter(|m| m.location.is_some())
+			.and_then(|m| m.location)
+			.and_then(|l| l.try_into().ok())
+	}
+}
+
+impl<ParachainId: Get<ParaId>, AssetRegistryInspector: AssetRegistryInspect> Convert<MultiLocation, Option<CurrencyId>> for CurrencyIdConvert<ParachainId, AssetRegistryInspector> {
+	fn convert(location: MultiLocation) -> Option<CurrencyId>  {
+		let para_id = ParachainId::get();
+
+		let unanchored_location = match location {
```

Comment:
The issue I found particularly was with xTokens::transfer. If in our registry we have the location from the local perspective, then the extrinsic will fail since apparently at the level of executing the message, the location must be from the external point of view. I couldn't find a way around this other than having a different currency convert just for this pallet. 

You are correct in the other scenarios that the location (for instance, an incoming transfer) will be anchored from our perspective given that that happens before the message arrives at our chain. 

And about the naming, it means unanchored from our perspective or at least that was the intention. It is essentially the external point of view, that of a sibling parachain.

Reply:
I am pretty sure that this is not how it should work. See last section [here](https://github.com/pendulum-chain/pendulum/pull/432#discussion_r1550658905).

Can you give a concrete example? I assume that this is about a reserve transfer deposit from our chain to another chain with an asset originating on our chain. What happens if you do the alternative call `pallet-xcm` → `limitedReserveTransferAssets` instead?

Reply:
I will try to test that extrinsic in our tests. We can continue with this discussion [here](https://github.com/pendulum-chain/pendulum/pull/432#discussion_r1551818616) since it is about the same and we can have all the information in one place. 
As a side note, I did try `xTokens::transfer_multiasset` which does allow to specify the multi location directly, and I see the same problem: `Unroutable`.